### PR TITLE
Enum refactor

### PR DESCRIPTION
**Pull request by t-keazirian**
[Trello ticket](https://trello.com/c/Pfbgppdl/51-refactor-constants-class)

- Added ENUM for HTTP Status Codes
- Added method to get message based on status code
- Refactored a few tests
- Added in methods to all handlers HTTP version, status code, status text

### CODE REVIEW

File: TKeazirian.HTTPServer/Handler/EchoBodyHandler.cs [hunk 1]
```diff
@@ -21,9 +27,19 @@ public Response HandleResponse(Request requestObject)
...
+
+    private string HandleResponseText(int responseStatusCode)
+    {
+        return GetStatusMessage(responseStatusCode);
+    }
+
+    public string HandleStatusLine(string httpVersion, int responseStatusCode, string responseStatusText)
     {
-        return Constants.Status200;
+        return httpVersion + Constants.Space + responseStatusCode + Constants.Space + responseStatusText;
```
Comment (Aoxian) [inline TKeazirian.HTTPServer/Handler/EchoBodyHandler.cs:42]: there is nothing specific to this Handler here, and this `HandleStatusLine` code could probably be shared across handlers? Could that be moved into the Interface? (it may need to change into an Abstract Class).

File: TKeazirian.HTTPServer/Constants.cs [hunk 2]
```diff
@@ -5,8 +5,9 @@ public static class Constants
         public const string NewLine = "\r\n";
         public const string BodySeparator = "\r\n\r\n";
         public const string Space = " ";
-        public const string Status200 = "HTTP/1.1 200 OK";
-        public const string Status404 = "HTTP/1.1 404 Not Found";
-        public static string Status301 = "HTTP/1.1 301 Moved Permanently";
+        public const string HttpVersion = "HTTP/1.1";
+        public const string Ok = "OK";
+        public const string NotFound = "Not Found";
+        public const string Moved = "Moved Permanently";
```
Comment (Aoxian) [inline TKeazirian.HTTPServer/Constants.cs:11]: You introduce an `enum` for [HttpStatusCode](https://github.com/t-keazirian/HTTPServer/blob/84a9780b8f83a3f90ee876033851562e50c912ec/TKeazirian.HTTPServer/Response/HttpStatusCode.cs#L1-L8); these three "constants" appear to be a mapping from `HttpStatusCode` to `HttpStatusMessage` ... does it make sense to extract an `HttpStatusMessage` as well?  or just keep them all in this `Constants` class?

File: TKeazirian.HTTPServer/Response/HttpStatusCode.cs [hunk 3]
```diff
@@ -0,0 +1,8 @@
+namespace TKeazirian.HTTPServer.Response;
+
+public enum HttpStatusCode
+{
+    Ok = 200,
+    Moved = 301,
+    NotFound = 404
+}
```
Comment (Aoxian) [inline TKeazirian.HTTPServer/Response/HttpStatusCode.cs:8]: This looks great. A quick glance, and you know exactly what's going on here.

File: TKeazirian.HTTPServer/Constants.cs (hunk 2 above)
Comment (Aoxian) [reply TKeazirian.HTTPServer/Constants.cs:11]: It actually looks like you have done over half the work in this [StatusMessage](https://github.com/t-keazirian/HTTPServer/blob/84a9780b8f83a3f90ee876033851562e50c912ec/TKeazirian.HTTPServer/Response/StatusMessage.cs#L1-L23) class.

File: TKeazirian.HTTPServer/Response/StatusMessage.cs [hunk 4]
```diff
@@ -0,0 +1,23 @@
+namespace TKeazirian.HTTPServer.Response;
+
+public static class StatusMessage
+{
+    public static string GetStatusMessage(int statusCode)
+    {
+        string message;
+        switch (statusCode)
+        {
+            case (int)HttpStatusCode.Ok:
```
Comment (Aoxian) [inline TKeazirian.HTTPServer/Response/StatusMessage.cs:10]: why do you have to cast to `int` here?

File: TKeazirian.HTTPServer.Tests/Response/ResponseTests.cs [hunk 5]
```diff
@@ -10,15 +10,16 @@ public class ResponseTests
     [Fact]
     public void BuildNewResponseBuildsNewResponseObject()
     {
-        string responseStatus = Constants.Status200;
-        string responseHeaders = HelperFunctions.CreateTestResponseHeaders();
-        string responseBody = "Hello world";
+        const string responseStatus = "HTTP/1.1 200 OK";
+        var responseHeaders = HelperFunctions.CreateTestResponseHeaders();
+        const string responseBody = "Hello world";
```
Comment (Aoxian) [inline TKeazirian.HTTPServer.Tests/Response/ResponseTests.cs:15]: why were these changed to `const` `string`? Constants are typically declared at a class level. did you mean for them to be `static`?

File: TKeazirian.HTTPServer/Handler/EchoBodyHandler.cs (hunk 1 above)
Comment (Aoxian) [reply TKeazirian.HTTPServer/Handler/EchoBodyHandler.cs:42]: alternatively, could it be moved into the Response itself? looks like potentially the only thing you would need to know is the `HttpStatusCode` and the rest could be calculated.

File: TKeazirian.HTTPServer/Response/StatusMessage.cs (hunk 4 above)
Comment (t-keazirian) [reply TKeazirian.HTTPServer/Response/StatusMessage.cs:10]: `HttpStatusCode.Ok` will give me the string value.  Casting it to an `int` gives me the integer value.

File: TKeazirian.HTTPServer.Tests/Response/ResponseTests.cs (hunk 5 above)
Comment (t-keazirian) [reply TKeazirian.HTTPServer.Tests/Response/ResponseTests.cs:15]: This is just what the Rider IDE said - it was giving me the green underline squiggly and then prompted me to do this.  So I figured it was the most correct?

Comment (Aoxian) [reply TKeazirian.HTTPServer.Tests/Response/ResponseTests.cs:15]: it seems odd to me. I'd just ignore the squigglies as unless there is a more compelling reason, this seems odd.

File: TKeazirian.HTTPServer/Constants.cs (hunk 2 above)
Comment (t-keazirian) [reply TKeazirian.HTTPServer/Constants.cs:11]: I kept them in `Constants` because I'm still referring to them in the `StatusMessage` class, and using the constants in tests...but are you saying to complete this I would take them out of `Constants` altogether and manage them in the `StatusMessage` class alone?

Comment (t-keazirian) [reply TKeazirian.HTTPServer/Constants.cs]: I pulled them out of `Constants` and put them into the `StatusMessage` class as class variables.

File: TKeazirian.HTTPServer/Handler/EchoBodyHandler.cs (hunk 1 above)
Comment (t-keazirian) [reply TKeazirian.HTTPServer/Handler/EchoBodyHandler.cs:42]: I like the idea of moving it into the Response itself.  Do you happen to have 30 min to pair with me on this?  I think it's part of the OO programming that I'm having trouble grasping and I want to really understand this.

**Review by Aoxian — APPROVED** [review]
